src/languages/r.js

```diff
@@ -2,76 +2,148 @@
+            'cospi cummax cummin cumprod cumsum digamma dim dimnames' +
+            'emptyenv exp expression floor forceAndCall gamma gc.time' +
+            'globalenv Im interactive invisible is.array is.atomic is.call' +
+            'is.character is.complex is.double is.environment is.expression' +
+            'is.finite is.function is.infinite is.integer is.language' +
+            'is.list is.logical is.matrix is.na is.name is.nan is.null' +
+            'is.numeric is.object is.pairlist is.raw is.recursive is.single' +
+            'is.symbol lazyLoadDBfetch length lgamma list log max min' +
+            'missing Mod names nargs nzchar oldClass on.exit pos.to.env' +
+            'proc.time prod quote range Re rep retracemem return round' +
+            'seq_along seq_len seq.int sign signif sin sinh sinpi sqrt' +
+            'standardGeneric substitute sum switch tan tanh tanpi tracemem' +
+            'trigamma trunc unclass untracemem UseMethod xtfrm',
         },
         relevance: 0
       },
-      {
-        // hex value
-        className: 'number',
-        begin: "0[xX][0-9a-fA-F]+[Li]?\\b",
-        relevance: 0
-      },
-      {
-        // explicit integer
-        className: 'number',
-        begin: "\\d+(?:[eE][+\\-]?\\d*)?L\\b",
-        relevance: 0
-      },
-      {
-        // number with trailing decimal
-        className: 'number',
-        begin: "\\d+\\.(?!\\d)(?:i\\b)?",
-        relevance: 0
-      },
-      {
-        // number
-        className: 'number',
-        begin: "\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",
-        relevance: 0
-      },
-      {
-        // number with leading decimal
-        className: 'number',
-        begin: "\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",
-        relevance: 0
-      },
+
+      // Roxygen comments
+      hljs.COMMENT(
+        /#'/,
+        /$/,
+        {
+          contains: [
+            {
+              // Handle `@examples` separately to cause all subsequent code
+              // until the next `@`-tag on its own line to be kept as-is,
+              // preventing highlighting. This code is example R code, so nested
+              // doctags shouldn’t be treated as such. See
+              // `test/markup/r/roxygen.txt` for an example.
+              className: 'doctag',
+              begin: '@examples',
+              starts: {
+                contains: [
+                  { begin: /\n/ },
+                  {
+                    begin: /#'\s*(?=@[a-zA-Z]+)/,
+                    endsParent: true,
+                  },
+                  {
+                    begin: /#'\s?/,
+                    end: /$/,
+                    excludeBegin: true,
+                  }
+                ]
+              }
+            },
+            {
+              // Handle `@param` to highlight the parameter name following
+              // after.
+              className: 'doctag',
+              begin: '@param',
+              starts: {
```

Comment:
Was contains not adequate here?

Reply:
The purpose is to have a new lexeme *after* `@param`, not nested inside. Otherwise all the formatting (background, bold …) of `doctag` also applies to it.

Reply:
Hmmm, I'm not sure whether to be pedantic here or not, but most of our other grammars do not think of doctags like this but rather like:

```
/**
@param {string} name - the name of the user
*/
````

```
<comment>/**
<doctag>@param <type>{string}</type> <variable>name</variable></doctag> - the name of the user
*/</comment>
```

Ie, the doctag includes the tag and it's metadata.  And some of our themes take advantage of this with `.doctag .type, .doctag .variable`, etc...

I think perhaps what you want here is nesting and `variable` instead of `meta`. Thoughts?

Reply:
> And some of our themes take advantage of this with .doctag .type, .doctag .variable, etc...

Seems I may be a liar here (I can't find it), but the prior point still stands about how this is typically handled.  I'm open to be persuaded, but this feels like something we should sort out and be consistent across grammars - as long-term I wouldn't mind a more modular way to handle doctags in general that is perhaps multi-grammar if possible.

Reply:
> Ie, the doctag includes the tag and it's metadata.

Oh, in that case I will *definitely* change it.

> and `variable` instead of `meta`

That’s much better, yes. I overlooked that class.